#RESULT

考慮到誤差值，調整取樣數值，以解決在第四次以後的數值發散問題/
```
$ py diffn.py 
df(sin, pi/4)= 0.7070007199422376
dfn(sin, 0 pi/4)= 0.7071067811865476
dfn(sin, 1 pi/4)= 0.7070007199422376
dfn(sin, 2 pi/4)= -0.7068946746064139
dfn(sin, 3 pi/4)= -0.7067886451754084
dfn(sin, 4 pi/4)= 0.7066826316590169
dfn(sin, 5 pi/4)= 0.7065766327652175
dfn(sin, 6 pi/4)= -0.706470658422707
dfn(sin, 7 pi/4)= -0.7063633774336147
dfn(sin, 8 pi/4)= 0.7062631803376259
dfn(sin, 9 pi/4)= 0.7047745295023075
dfn(sin, 10 pi/4)= -0.7091680517951671
dfn(sin, 11 pi/4)= 0.7572824520493071
dfn(sin, 12 pi/4)= 2.7374344249189404
dfn(sin, 13 pi/4)= -1560.760545091262
dfn(sin, 14 pi/4)= -1088.939809198756
```
/
下為執行結果:/
可發現在第十三次執行後其結果仍然出現無法預期的錯誤/
```
dfn(sin, 12 pi/4)= 2.7374344249189404
```
